pkg/controller/migmigration/registry.go

```diff
@@ -23,14 +23,18 @@ func (t *Task) getAnnotations(client k8sclient.Client) (map[string]string, error
 	if registryService == nil {
 		return nil, errors.New("migration registry service not found")
 	}
-	registryDC, err := t.PlanResources.MigPlan.GetRegistryDC(client)
+	registryDC, err := t.PlanResources.MigPlan.GetRegistryDeployment(client)
 	if err != nil {
 		return nil, err
 	}
 	if registryDC == nil {
 		return nil, errors.New("migration registry DeploymentConfig not found")
 	}
 
+	if registryDC.DeletionTimestamp != nil {
+		return nil, errors.New(fmt.Sprintf("DeploymentConfig %s/%s is being garbage collected with deletion timestamp %s", registryDC.Namespace, registryDC.Name, registryDC.DeletionTimestamp))
```

Comment:
1. Is this meant to return an error? What is the consequence _to us_ if a `DeletionTimestamp` is set but has not yet been acted upon?

2. This error message should be changed to reflect that we are now using a `Deployment` for registry, not a `DeploymentConfig`.